WWIV5TelnetServer/WWIV5TelnetServer/CheckUpdates.cs

```diff
@@ -118,34 +119,74 @@ public void DoUpdateCheck(object source, ElapsedEventArgs e)
             {
                 updateTimer.AutoReset = false;
                 updateTimer.Enabled = false;
-                MessageBox.Show("A Newer Version of WWIV is Available!");
-                // TODO Launch WWIV Update For Windows Once Packaged with Distribution.
+                if (MessageBox.Show("A Newer Version of WWIV is Available!\r\n \r\nLaunch WWIV Update Now?", "UPDATE AVAILABLE", MessageBoxButtons.YesNo, MessageBoxIcon.Information) == DialogResult.Yes)
+                {
+                    // Launch WWIV Update
+                    ProcessStartInfo wwivUpdate = new ProcessStartInfo("wwiv-update.exe");
+                    Process.Start(wwivUpdate);
+                }
+                else
+                {
+                    // NO - Do Nothing...
+                }
             }
 
             // Periodical Update Checking
             if (UserUpdatePref == "Hourly" && newBuild > oldBuild && hourCount >= 1)
             {
                 updateTimer.Enabled = false;
-                MessageBox.Show("A Newer Version of WWIV is Available!");
-                // TODO Launch WWIV Update For Windows Once Packaged with Distribution.
+                if (MessageBox.Show("A Newer Version of WWIV is Available!\r\n \r\nLaunch WWIV Update Now?", "UPDATE AVAILABLE", MessageBoxButtons.YesNo, MessageBoxIcon.Information) == DialogResult.Yes)
+                {
+                    // Launch WWIV Update
+                    ProcessStartInfo wwivUpdate = new ProcessStartInfo("wwiv-update.exe");
+                    Process.Start(wwivUpdate);
+                }
+                else
```

Comment:
I'm not seeing what you did here but I do play to minimize the code for each scheduled option in the next release.